README.md

```diff
@@ -74,16 +74,27 @@ dot2, err := dotsql.LoadFromFile("queries2.sql")
 dot := dotsql.Merge(dot1, dot2)
 ```
 
+Text Interpolation
+--
+[text/template](https://pkg.go.dev/text/template)-style text interpolation is supported.
+
+To use, call `.WithData(any)` on your dotsql instance to
+create a new instance which passes those values into the templating library.
+
+```sql
+-- name: count-users
+SELECT count(*) FROM users {{if .exclude_deleted}}WHERE deleted IS NULL{{end}}
+```
+
+```go
+dotsql.WithData(map[string]any{"exclude_deleted": true}).Query(db, "count-users")
```

Comment:
did you consider the case where 2 queries rely on a variable named `exclude_deleted`  but they need 2  different values? Would it make sense that `WithData` treats the `string` key of `map[string]any` as the query name?

Reply:
So I think there the user of this has a couple ways they can do it.

The most flexible would be to use separate objects:

`dotsql.WithData(map[string]any{"exclude_deleted": true}).Query(db, "count-users")`
and
`dotsql.WithData(map[string]any{"exclude_deleted": func() bool { return userConfig.ExcludeDeleted }}).Query(db, "list-products")`

Each of those results in a separate instance of a `dotsql` with different data attached to them. Up to the user to pick the appropriate one.

Or just name your data distinctly.

Reply:
I didn't see that, it makes a lot of sense!